### PR TITLE
Add initial support for RESTful unit-tests via WireMock

### PR DESCRIPTION
**Pull request by dgiagio**
- Add WireMock jar and slf4j minimal dependency
- Add unit-test for /login using WireMock.

### CODE REVIEW

**Comment by renner** [comment]
Would you mind to give us a bit of reasoning about why you think we should use especially the WireMock library for testing? We want to know your opinion, so maybe you could explain your decision by answering the following questions:

- Why is WireMock better than other available libraries?
- When comparing it to e.g. [REST-driver] (https://github.com/rest-driver/rest-driver/wiki) (that has been proposed as well), what are the advantages or disadvantages?
- Is there any other reasons why you think we should use WireMock?

**Comment by dgiagio** [comment]
I chose WireMock mostly because:
- It's one of the most popular libraries for HTTP mocking, it's actively developed and well-documented.
- WireMock is more popular than REST-driver (370+ stars vs 90+ stars on GitHub), has many more features (stub delays for timeout simulation, stateful behavior, can be used to record real service responses to build stubs, and so on), much better documentation and better API design in my opinion.
- My main goal was to pick a mock library that has enough features and good documentation. I think WireMock fits the bill.

On other hand, I have no problem using other library if you guys decide so.

**Comment by renner** [comment]
Thanks a lot for your reasoning! I like the library and this PR, even though I might prefer to verify our requests separately from the responses which can be done like [this] (http://wiremock.org/verifying.html) it seems. Please find some additional inline comments as well.

Note that there is no hurry, I will wait with merging this to see the proposal by @skhtmp as well.

File: src/test/java/com/suse/saltstack/netapi/utils/SaltStackClientTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package com.suse.saltstack.netapi.client;
```
Comment (renner) [inline src/test/java/com/suse/saltstack/netapi/utils/SaltStackClientTest.java]: This package does not match the file system path above, please fix the path in this case.

**Comment by dgiagio** [comment]
@renner You're correct, the file system path was incorrect. Fixed, thanks!

**Comment by renner** [comment]
We eventually decided to use WireMock, so I merged this. Thanks for your contribution!

**Comment by dgiagio-druid** [comment]
Thanks!

**Pull request merged**